guardrails/validator_base.py

```diff
@@ -231,10 +232,11 @@ class FailResult(ValidationResult):
     fix_value: Optional[Any] = None
 
 
+@dataclass  # type: ignore
```

Comment:
In pydantic 2.x this causes Validators to serialize to `{ "rail_alias": "{value}" }`.  This is more of a workaround than a fix, but it allows us to utilize function calling on models with properties that are also models.